hash/leapster.xml

```diff
@@ -287,44 +324,56 @@ Known games listed by part-no, (*) denotes undumped:
 		<publisher>LeapFrog</publisher>
 		<info name="serial" value="500-13448-A" />
 		<part name="cart" interface="leapster_cart">
+			<feature name="pcb" value="MX64Mb" />
+			<feature name="u2" value="U2" /> <!-- 24LC16B -->
+			<feature name="u3" value="U3" /> <!-- EPOXY BLOB ROM -->
 			<dataarea name="rom" size="8388608">
-				<rom name="500-13448-a.bin" size="8388608" crc="2465f01d" sha1="d37d7ea21cf8ed01ecf497c3c99185529aee59af" offset="0" />
+				<rom name="500-13448-a.u3" size="8388608" crc="2465f01d" sha1="d37d7ea21cf8ed01ecf497c3c99185529aee59af" offset="0" />
 			</dataarea>
 		</part>
 	</software>
 
 	<software name="dora" supported="no">
-		<description>Dora - Retter der Wildnis (Ger)</description>
+		<description>Nick Jr Dora - Retter der Wildnis (Ger)</description>
 		<year>2003</year>
 		<publisher>LeapFrog</publisher>
 		<info name="serial" value="500-12562-B" />
 		<part name="cart" interface="leapster_cart">
+			<feature name="pcb" value="FAB-300-1XXXX-A" />
+			<feature name="u1" value="U1" /> <!-- EPOXY BLOB ROM -->
+			<feature name="u2" value="U2" /> <!-- 24LC16B -->
 			<dataarea name="rom" size="8388608">
-				<rom name="500-12562-b.bin" size="8388608" crc="242907d4" sha1="262ee3e208fff0fc362bc879416cbdf31bf38304" offset="0" />
+				<rom name="500-12562-b.u1" size="8388608" crc="242907d4" sha1="262ee3e208fff0fc362bc879416cbdf31bf38304" offset="0" />
 			</dataarea>
 		</part>
 	</software>
 
 	<software name="findnemo" supported="no">
-		<description>Finding Nemo (USA)</description>
+		<description>Disney/Pixar Finding Nemo (USA)</description>
 		<year>2005?</year>
 		<publisher>LeapFrog</publisher>
 		<info name="serial" value="500-10775-A" />
 		<part name="cart" interface="leapster_cart">
+			<feature name="pcb" value="FAB-300-1XXXX-A" />
+			<feature name="u1" value="U1" /> <!-- EPOXY BLOB ROM -->
+			<feature name="u2" value="U2" /> <!-- 24LC02B -->
 			<dataarea name="rom" size="8388608">
-				<rom name="500-10775-a.bin" size="8388608" crc="1895f946" sha1="8baf4f44f64d17333a82ce29e59c62e51e45c0c0" offset="0" />
+				<rom name="500-10775-a.u1" size="8388608" crc="1895f946" sha1="8baf4f44f64d17333a82ce29e59c62e51e45c0c0" offset="0" />
 			</dataarea>
 		</part>
 	</software>
 
 	<software name="findnemog" cloneof="findnemo" supported="no">
-		<description>Findet Nemo (Ger)</description>
+		<description>Disney/Pixar Findet Nemo (Ger)</description>
 		<year>2003</year>
 		<publisher>LeapFrog</publisher>
 		<info name="serial" value="500-11495-A" />
 		<part name="cart" interface="leapster_cart">
+			<feature name="pcb" value="FAB-300-10028-02" />
+			<feature name="u2" value="U2" /> <!-- 24LC02B -->
+			<feature name="u3" value="U3" /> <!-- EPOXY BLOB ROM -->
```

Comment:
Can you explain what value these are supposed to provide, with the name and value basically being the same thing?  There's no way a driver could use it to determine correct mapping/mirroring or anything.  I don't want to see `feature` attributes turn into the same stupid mess we have with ROM labels in the main source - each software list needs a clear, documented schema for the features in an obvious place, like the header comment of the file.

Reply:
Oh yeah, you remind me of a thing I forgot to ask. These carts obviously have an EEPROM on the board, but no documentation of any `nvram` in the `dataarea`. Should I add that? The carts seem to have different size nvram per game, so it makes sense to me.

This tells you that the cart has a `FAB-300-10028-02` board with two IC slots on it. `U1` for some reason has been left out in manufacture on this cart's board. There are two other slots on the board, marked `U2` and `U3`. `U2` has a 2K EEPROM while `U3` has the ROM, in COB epoxy form. I'm basically just doing what many other software lists are doing.

Reply:
Well, you need to agree on a schema for it with the driver developer.  It should be theoretically possible to parse the `feature` elements to configure emulation for the cartridge, even if the driver doesn't implement it.  This isn't even a step in that direction.  For now, just add comments.